rest_sensors.cpp

```diff
@@ -906,45 +906,59 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                 else if (rid.suffix == RConfigOffset) // Signed integer
                 {
                     data.integer = data.integer / 10;
-
-                    if ((R_GetProductId(sensor) == QLatin1String("Tuya_THD HY369 TRV") ||
-                         R_GetProductId(sensor) == QLatin1String("Tuya_THD Essentials TRV") ||
-                         R_GetProductId(sensor) == QLatin1String("Tuya_THD Smart radiator TRV") ||
-                         R_GetProductId(sensor) == QLatin1String("Tuya_THD NX-4911-675 TRV") ||
-                         R_GetProductId(sensor) == QLatin1String("Tuya_THD SEA801-ZIGBEE TRV")) ||
-                         R_GetProductId(sensor) == QLatin1String("Tuya_THD WZB-TRVL TRV"))
+                    
+                    if (R_GetProductId(sensor) == QLatin1String("Tuya_THD HY369 TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD HY368 TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD Essentials TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD Smart radiator TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD NX-4911-675 TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD SEA801-ZIGBEE TRV") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD BTH-002 Thermostat") ||
+                        R_GetProductId(sensor) == QLatin1String("Tuya_THD WZB-TRVL TRV"))
                     {
                         QByteArray tuyaData;
                         bool alternative = false;
+                        
+                        qint8 offset2 = data.integer / 10;
```

Comment:
Just saw the `qint8`, it  can only hold -127..127.
Later on 4096 is added, why not just use `int`?

Reply:
I don't understand how the code have worked before.

>23:54:20:989 Send Tuya request 0x50325FFFFE289FB3 : Dp_type: 0x02, Dp_identifier 0x1B, data: fffffffd

How I can have qint32 value like 0xfffffffd with a qint8 base ...

Reply:
>How I can have qint32 value like 0xfffffffd with a qint8 base ...

You can't ;) in `(offset2 >> 24)` the `qint8` aka `signed char`  is *promoted* to `int`.

See: https://www.geeksforgeeks.org/integer-promotions-in-c

But this is usually not what you want and might result in all sorts of unexpected outcomes. It's better and easier to reason about to just use a type which can hold all required values, like here `int`.

Reply:
>like here int

Hu, not a qint32 ?
I need a signed value, because some device use negative value and coded on 32 bit ? for exemple for -3 0xfffffffd

>You can't ;) in (offset2 >> 24) the qint8 aka signed char is promoted to int

But it s that I don't understand, it have worked the log was with the actual code
>23:54:20:989 Send Tuya request 0x50325FFFFE289FB3 : Dp_type: 0x02, Dp_identifier 0x1B, data: fffffffd

Reply:
> Hu, not a qint32 ?

They are the same, qint32 is just a typedef for int.

`typedef int qint32;                /* 32 bit signed */`

> I need a signed value, because some device use negative value and coded on 32 bit ? for exemple for -3 0xfffffffd

I haven't looked too deep into the protocol yet, but assume that `data` is just an signed 32 bit integer.
Therefore `int` or `qint32` is the suitable type to use.

> In my mind (offset >> 24) had to give 0x00 and not 0xFF

Not for *signed* integers, in binary these are represented in [two's-complement](https://www.tutorialspoint.com/two-s-complement).

0xFFFFFFFD  is  -3